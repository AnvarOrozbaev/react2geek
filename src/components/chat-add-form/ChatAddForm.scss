$primary:   #864f12;
.add-chat{
  display:flex;
  align-items: center;
  height: 2rem;
&__input{
  border-radius: 10px;
  border: 1px solid $primary;
  border-right: none;
  border-top-right-radius: 0 ;
  border-bottom-right-radius: 0 ;
  outline: none;
  height: 1.2em;
}
&__btn{
  border-radius: 10px;
  border: 1px solid $primary;
  border-top-left-radius: 0 ;
  border-bottom-left-radius: 0 ;
  background-color: $primary;
  color: #ffffff;
  &:hover{
    background-color: #ffffff;
    color: $primary;
  }
  
}
}