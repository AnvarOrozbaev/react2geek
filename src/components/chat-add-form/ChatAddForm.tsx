import React, { FC, useState } from 'react';
import './ChatAddForm.scss';

interface ChatListddFormProps {
  handleAddChat: (name: string) => void;
}
export const ChatAddForm: FC<ChatListddFormProps> = ({ handleAddChat }) => {
  const [name, setName] = useState('');
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>, name: string) => {
    e.preventDefault();
    if (name.length === 0) {
      return;
    }
    handleAddChat(name);
    setName('');
  };

  return (
    <form
      className="add-chat"
      onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e, name)}
    >
      <input
        value={name}
        className="add-chat__input"
        placeholder="название чата"
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          setName(e.target.value)
        }
      />
      <input
        type="submit"
        value="+"
        className="add-chat__btn"
        data-testid="add-chat__btn"
      />
    </form>
  );
};
