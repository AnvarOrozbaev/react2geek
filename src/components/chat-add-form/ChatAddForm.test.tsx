import React from 'react';
import { ChatAddForm } from './ChatAddForm';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('ChatAddForm', () => {
  const mockAddChat = jest.fn();

  it('render component', () => {
    render(<ChatAddForm handleAddChat={mockAddChat} />);
  });
  it('render with snapshot', () => {
    const { asFragment } = render(<ChatAddForm handleAddChat={mockAddChat} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('form behavior when input is empty', async () => {
    const { getByTestId, getByPlaceholderText } = render(
      <ChatAddForm handleAddChat={mockAddChat} />
    );
    const btn = getByTestId('add-chat__btn');
    const input = getByPlaceholderText(/название чата/i);
    fireEvent.change(input, { target: { value: '' } });
    fireEvent.click(btn);
    expect(mockAddChat).toHaveBeenCalledTimes(0);
  });
  it('form submit', async () => {
    const { getByTestId, getByPlaceholderText } = render(
      <ChatAddForm handleAddChat={mockAddChat} />
    );
    const btn = getByTestId('add-chat__btn');
    const input = getByPlaceholderText(/название чата/i);
    fireEvent.change(input, { target: { value: 'chat-4' } });
    fireEvent.click(btn);
    expect(mockAddChat).toBeCalled();
  });
});
