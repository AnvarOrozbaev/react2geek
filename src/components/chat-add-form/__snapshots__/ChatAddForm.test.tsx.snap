// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatAddForm render with snapshot 1`] = `
<DocumentFragment>
  <form
    class="add-chat"
  >
    <input
      class="add-chat__input"
      placeholder="название чата"
      value=""
    />
    <input
      class="add-chat__btn"
      data-testid="add-chat__btn"
      type="submit"
      value="+"
    />
  </form>
</DocumentFragment>
`;
